pkg/fetcher/fetcher.go

```diff
@@ -401,6 +406,19 @@ func (u *URLFetcher) IsStatusNotFound() bool {
 	return u.StatusCode == http.StatusNotFound
 }
 
+// IsNonretryableClientError returns true if status code is a nonretryable 4XX error. This includes
+// all 4XX errors except 'request timeout', 'locked', and 'too many requests'. Additionally, it
+// does not return 'not found' or 'unauthorized' errors since those are handled uniquely.
+func (u *URLFetcher) IsNonretryableClientError() bool {
+	s := u.StatusCode
+	return 400 <= s && s < 500 &&
+		s != http.StatusNotFound &&
+		s != http.StatusUnauthorized &&
+		s != http.StatusRequestTimeout &&
```

Comment:
Ah, I misread the description of status 408. StatusRequestTimeout should retry upon failure.